### PR TITLE
decomp: `entity-h`, `camera-h`, `cam-update-h` and `cam-interface[-h]` 

### PR DESCRIPTION
**Pull request by xTVaser**
Opening for visibility, just waiting on `entity-camera` and `entity-ambient`.  Which are super weird types, they have no inspect/print methods that i could find at all, so it doesn't seem like we missed them.

### CODE REVIEW

File: goal_src/engine/entity/entity-h.gc [hunk 1]
```diff
@@ -5,3 +5,206 @@
...
+  :size-assert         #x18
+  :flag-assert         #x900000018
+  )
+
+;; NOTE - this is a strange pattern...this symbol isn't defined until a later file 'navigate'
+;; But this seems to be setting the symbol to nothing if it's not found, but of course, our compiler freaks out
+(define-extern entity-nav-login function)
+(if (zero? entity-nav-login)
+  (set! entity-nav-login nothing)
+  )
```
Comment (xTVaser) [inline goal_src/engine/entity/entity-h.gc:189]: calling this out

File: goal_src/engine/camera/cam-interface.gc [hunk 2]
```diff
@@ -5,3 +5,126 @@
...
+   (set! *camera-dummy-vector* v1-8)
+   )
+  )
+
+;; definition for function camera-pos
+;; INFO: Return type mismatch object vs vector.
+(defun camera-pos ()
+  (the-as vector (cond
+                  (*camera-combiner*
+                   (-> *camera-combiner* stack)
```
Comment (water111) [inline goal_src/engine/camera/cam-interface.gc:81]: https://github.com/water111/jak-project/issues/467 strikes again!  Fine for now, just noting this for when I try to fix this.

File: goal_src/engine/camera/cam-interface.gc [hunk 3]
```diff
@@ -5,3 +5,126 @@
...
+      )
+     )
+    (.add.x.vf vf1 vf0 vf0 :mask #b1000)
+    (.mul.x.vf vf1 vf1 vf2 :mask #b111)
+    (.svf (&-> v1-0 quad) vf1)
+    )
+   (vector+float*!
+    arg0
+    arg0
+    (the-as vector (&-> *math-camera* inv-camera-rot data 4))
```
Comment (water111) [inline goal_src/engine/camera/cam-interface.gc:37]: https://github.com/water111/jak-project/issues/467 again

File: goal_src/engine/camera/cam-interface.gc [hunk 4]
```diff
@@ -5,3 +5,126 @@
...
+        )
+   (atan f1-0 f0-0)
+   )
+  )
+
+;; definition for function camera-teleport-to-entity
+;; INFO: Return type mismatch int vs none.
+;; Used lq/sq
+(defun camera-teleport-to-entity ((arg0 entity-actor))
+  (local-vars (pp process))
```
Comment (water111) [inline goal_src/engine/camera/cam-interface.gc:111]: https://github.com/water111/jak-project/issues/493

**Pull request merged**